src/edit_view.rs

```diff
@@ -139,74 +254,85 @@ impl EditView {
         })
     }
 
-    pub fn set_view_id(&mut self, view_id: &str) {
-        self.view_id = view_id.into();
-    }
-
     pub fn apply_update(&mut self, update: &Value) {
         self.line_cache.apply_update(update);
         self.constrain_scroll();
     }
 
-    pub fn char(&self, ch: u32, _mods: u32, win: &MainWin) {
+    pub fn char(&self, ch: u32, _mods: u32) {
         if let Some(c) = ::std::char::from_u32(ch) {
             if ch >= 0x20 {
                 // Don't insert control characters
                 let params = json!({"chars": c.to_string()});
-                self.send_edit_cmd("insert", &params, win);
+                self.send_edit_cmd("insert", &params);
             }
         }
     }
 
-    fn send_edit_cmd(&self, method: &str, params: &Value, win: &MainWin) {
-        win.send_edit_cmd(method, params, &self.view_id);
+    fn send_edit_cmd(&self, method: &str, params: &Value) {
+        let edit_params = json!({
+            "method": method,
+            "params": params,
+            "view_id": &self.view_id,
+        });
+        self.send_notification("edit", &edit_params);
+    }
+
+    fn send_notification(&self, method: &str, params: &Value) {
+        if let Some(ref core) = self.core.upgrade() {
+            core.lock().unwrap().send_notification(method, params);
+            println!("fe->core: {}", json!({
```

Comment:
This is a bit spammy, but ok for now. What I usually do is leave it in with a comment, so it's handy to turn on when debugging.

(Many style guides frown on commented-out code, but in this particular case I think it makes sense)

Reply:
Ah. I'll comment it out for now then. (I think it would be nice if we could just use trace logging instead at some point)